pallets/market/src/lib.rs

```diff
@@ -1041,8 +873,8 @@ pub mod pallet {
                 Self::validate_deal_can_activate(deal, provider, sector_expiry, sector_activation)
                     .map_err(|e| {
                         log::error!(target: LOG_TARGET, "deal {} cannot be activated, because: {:?}", *deal_id, e);
-                        Error::<T>::DealActivationError }
-                    )?;
+                        e
```

Comment:
There is `inspect_err`

Reply:
Adding `inspect_err` here makes it more confusing IMO.